modules/tinymce/src/plugins/link/main/ts/ui/DialogInfo.ts

```diff
@@ -20,7 +20,7 @@ const nonEmptyAttr = (dom: DOMUtils, elem: string | Element, name: string): Opti
 const extractFromAnchor = (editor: Editor, anchor: HTMLAnchorElement): LinkDialogInfo['anchor'] => {
   const dom = editor.dom;
   const onlyText = Utils.isOnlyTextSelected(editor);
-  const text: Optional<string> = onlyText ? Optional.some(Utils.getAnchorText(editor.selection, anchor)) : Optional.none();
+  const text: Optional<string> = onlyText ? Optional.some(Utils.getAnchorText(editor.selection, Optional.from(anchor))) : Optional.none();
```

Comment:
Why are we doing `Optional.from(anchor)` here if it's not nullable (or at least according to the types)? We should be passing an Optional here instead if this can in fact be nullable. I suspect this is also why you're doing `anchorNode.getOrNull()` in `Dialog.ts` which is normally a bit of a red flag.

- [ ] Make this function take an `Optional<HTMLAnchorElement>` and pass it through from `Dialog.ts` if it can in fact be nullable

Reply:
Done!